src/Microsoft.Azure.WebJobs.Extensions.Kafka/Output/KafkaProducerAsyncCollector.cs

```diff
@@ -32,14 +34,60 @@ public Task AddAsync(T item, CancellationToken cancellationToken)
 
             object messageToSend = item;
 
-            if (item.GetType() == typeof(string) || item.GetType() == typeof(byte[]))
+            if (item.GetType() == typeof(string))
+            {
+                messageToSend = convertToKafkaEventData(item);
+            }
+
+            if (item.GetType() == typeof(byte[]))
             {
                 messageToSend = new KafkaEventData<T>(item);
             }
 
             return entity.SendAndCreateEntityIfNotExistsAsync(messageToSend, functionInstanceId, cancellationToken);
         }
 
+        private object convertToKafkaEventData(T item)
+        {
+            try
+            {
+                return buildKafkaDataEvent(item);
+            }
+            catch (Exception)
+            {
+                return new KafkaEventData<T>(item);
+            }
+        }
+
+        private object buildKafkaDataEvent(T item)
+        {
+            JObject dataObj = JObject.Parse(item.ToString());
```

Comment:
You can use .FromObject method https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_Linq_JObject_FromObject.htm

Reply:
I tried this and it didn't work so moved back to Parse function only